client/Command/ListSnapshots/SnappersData/Table.cc

```diff
@@ -0,0 +1,270 @@
+    {
+
+	Command::ListSnapshots::SnappersData::Table::Table(
+	    const Command::ListSnapshots& command, TableLineStyle style) :
+	    SnappersData(command), _style(style)
+	{
+	    calculate_used_space();
+	}
+
+
+	vector<Command::ListSnapshots::SnappersData::Attribute>
+	Command::ListSnapshots::SnappersData::Table::all_mode_attributes() const
+	{
+	    return { NUMBER, TYPE, PRE_NUMBER, DATE, USER, USED_SPACE, CLEANUP, DESCRIPTION, USERDATA };
+	}
+
+
+	vector<Command::ListSnapshots::SnappersData::Attribute>
+	Command::ListSnapshots::SnappersData::Table::single_mode_attributes() const
+	{
+	    return { NUMBER, DATE, USER, USED_SPACE, DESCRIPTION, USERDATA };
+	}
+
+
+	vector<Command::ListSnapshots::SnappersData::Attribute>
+	Command::ListSnapshots::SnappersData::Table::pre_post_mode_attributes() const
+	{
+	    return { NUMBER, POST_NUMBER, DATE, POST_DATE, DESCRIPTION, USERDATA };
+	}
+
+
+	string Command::ListSnapshots::SnappersData::Table::attribute_label(
+	    const Command::ListSnapshots::SnappersData::Attribute attribute) const
+	{
+	    switch (attribute)
+	    {
+		case CONFIG:
+		    return _("Config");
+
+		case SUBVOLUME:
+		    return _("Subvolume");
+
+		case NUMBER:
+		    return number_label();
+
+		case DEFAULT:
+		    return _("Default");
+
+		case ACTIVE:
+		    return _("Active");
+
+		case TYPE:
+		    return _("Type");
+
+		case DATE:
+		    return date_label();
+
+		case USER:
+		    return _("User");
+
+		case USED_SPACE:
+		    return _("Used Space");
+
+		case CLEANUP:
+		    return _("Cleanup");
+
+		case DESCRIPTION:
+		    return _("Description");
+
+		case USERDATA:
+		    return _("Userdata");
+
+		case PRE_NUMBER:
+		    return _("Pre #");
+
+		case POST_NUMBER:
+		    return _("Post #");
+
+		case POST_DATE:
+		    return _("Post Date");
+
+		default:
+		    throw std::logic_error("invalid attribute");
+	    }
+	}
+
+
+	string Command::ListSnapshots::SnappersData::Table::number_label() const
+	{
+	    if (list_mode() == LM_PRE_POST)
+		return _("Pre #");
+
+	    return _("#");
+	}
+
+
+	string Command::ListSnapshots::SnappersData::Table::date_label() const
+	{
+	    if (list_mode() == LM_PRE_POST)
+		return _("Pre Date");
+
+	    return _("Date");
+	}
+
+
+	string Command::ListSnapshots::SnappersData::Table::number_value(const ProxySnapshot& snapshot,
+	    const ProxySnapshots& snapshots) const
+	{
+	    bool is_default = is_default_snapshot(snapshot, snapshots);
+	    bool is_active = is_active_snapshot(snapshot, snapshots);
+
+	    static const char sign[2][2] = { { ' ', '-' }, { '+', '*' } };
+
+	    return Command::ListSnapshots::SnappersData::number_value(snapshot, snapshots) +
+		sign[is_default][is_active];
+	}
+
+
+	string
+	Command::ListSnapshots::SnappersData::Table::snapshot_date(const ProxySnapshot& snapshot) const
+	{
+	    bool utc = command().global_options().has_utc_option();
+	    bool iso = command().global_options().has_iso_option();
+
+	    return datetime(snapshot.getDate(), utc, iso);
```

Comment:
@mvidner @joseivanlopez the output of this has the text translated and we don't want that for machine-readable outputs

Reply:
Output is translated when not using --iso option. Here is correct, because is the value for the table output. For machine-readable outputs, iso output is always shown.